### PR TITLE
node: don't set write-only NodeAddressingElement.AddressType property

### PR DESCRIPTION
**Pull request by tklauser**
This field is only ever set but never read (i.e. inside the CNI). We
need to keep the property as part of the API because of backwards
compatibility, but it's fine to no longer set it.
    
This allows to drop `pkg/node.getNodeIP` and fold it into
`pkg/node.GetNodeIP`.

### CODE REVIEW

File: api/v1/openapi.yaml [hunk 1]
```diff
@@ -2327,9 +2327,6 @@ definitions:
       alloc-range:
         description: Address pool to be used for local endpoints
         type: string
-      address-type:
-        description: Node address type, one of HostName, ExternalIP or InternalIP
-        type: string
```
Comment (rolinh) [inline api/v1/openapi.yaml]: In [the doc](https://docs.cilium.io/en/latest/api/#compatibility-guarantees), we make the following statement:
> Cilium API is stable as of version 1.0, backward compatibility will be upheld for whole lifecycle of Cilium 1.x.

My understanding is that this change breaks this promise. Could we somehow keep this but make this a no-op and mark deprecated?

**Review by rolinh — CHANGES_REQUESTED** [review]

File: api/v1/openapi.yaml (hunk 1 above)
Comment (tklauser) [reply api/v1/openapi.yaml]: Right, good point. I seem to remember we had ro revert other changes like this due to failures to unmarshal older versions of certain definitions.

I guess we should keep the field in the API and just no longer set it (i.e. keep this PR's changes to `pkg/node`, but leave the API untouched).

**Review by rolinh — APPROVED** [review]

**Review by jibi — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Pull request merged**